lib/hackney/cloud/adapter/aws_s3.rb

```diff
@@ -2,18 +2,40 @@ module Hackney
   module Cloud
     module Adapter
       class AwsS3
-        def initialize(stub: false)
-          client = Aws::S3::Client.new(stub_responses: stub)
+        def initialize(encryption_client: nil, stub: false)
+          # Create KMS client
+          kms = Aws::KMS::Client.new(stub_responses: stub)
 
-          @s3 = Aws::S3::Resource.new(client: client)
+          # Create encryption client
+          @client = encryption_client || Aws::S3::Encryption::Client.new(
+            kms_key_id: customer_managed_key,
+            kms_client: kms
+          )
         end
 
-        def upload(bucketname, filename, new_filename)
-          s3_file_obj = @s3.bucket(bucketname).object(new_filename)
+        def upload(bucket_name, filename, new_filename)
+          contents = File.read(filename)
```

Comment:
minor thing, but I'm guessing this will just be a singular `content` of the file not multiple contents
```suggestion
          content = File.read(filename)
```